### PR TITLE
feat: Remove User endpoint.

### PR DESCRIPTION
**Pull request by magajh**
This feature adds an endpoint to safely remove an account from the platform.
This new endpoint is intended to be used in the CLI.

### CODE REVIEW

File: eox_core/api/v1/views.py [hunk 1]
```diff
@@ -200,6 +211,93 @@ def patch(self, request, *args, **kwargs):
         return Response(serialized_user.data)
 
 
+class EdxappRemoveUser(UserQueryMixin, APIView):
```
Comment (MoisesGSalas) [inline eox_core/api/v1/views.py:214]: This view has quite a bit of business logic attached to it. What if we move that to the users backend? Similar to [`create_edxapp_user`](https://github.com/eduNEXT/eox-core/blob/6080576f28557675e41e1c3964842894063288fc/eox_core/edxapp_wrapper/backends/users_j_v1.py#L63)

File: eox_core/api/v1/views.py [hunk 2]
```diff
@@ -200,6 +211,93 @@ def patch(self, request, *args, **kwargs):
...
+    renderer_classes = (JSONRenderer, BrowsableAPIRenderer)
+
+    def delete(self, request, *args, **kwargs):  # pylint: disable=too-many-locals
+        """
+        Allows to safely remove an edxapp User.
+
+        For example:
+
+        **Requests**:
+            DELETE <domain>/eox-core/api/v1/remove-user/
```
Comment (MoisesGSalas) [inline eox_core/api/v1/views.py:230]: Why don't we use the existing `/eox-core/api/v1/user/`?

Comment (magajh) [reply eox_core/api/v1/views.py:230]: @MoisesGSalas I was unsure about where to put it. I thought that maybe later we would want to have exclusive permissions for these endpoints that are intended to be used on the CLI, that's why I put it in a different view

Comment (MoisesGSalas) [reply eox_core/api/v1/views.py:230]: In that case what about an endpoint for CLI usage? maybe we can consider this a different API as it's meant to be used by the support team?

Comment (magajh) [reply eox_core/api/v1/views.py]: @MoisesGSalas done, I added the support API

File: eox_core/api/v1/views.py (hunk 1 above)
Comment (magajh) [reply eox_core/api/v1/views.py]: done

**Review by MoisesGSalas — COMMENTED** [review]
Can you add some tests?

**Comment by magajh** [comment]
@MoisesGSalas I didn't add tests for this view since the deletion of the user is made completely in the user's backend by calling different methods of the platform

File: eox_core/edxapp_wrapper/backends/users_j_v1.py [hunk 3]
```diff
@@ -180,6 +190,74 @@ def get_edxapp_user(**kwargs):
...
+                    )
+                    status_code = status.HTTP_200_OK
+
+    msg = "{user} does not have a signup source on the site {site}".format(user=user_response, site=site)
+    status_code = status.HTTP_400_BAD_REQUEST
+    response_dict = {
+        "status": status_code,
+        "message": msg,
+    }
+    return json.dumps(response_dict)
```
Comment (MoisesGSalas) [inline eox_core/edxapp_wrapper/backends/users_j_v1.py:258]: Now that I see this, is there any need for the json dump? wouldn't the `Response` object take care of that?

**Comment by MoisesGSalas** [comment]
@magajh 
> @MoisesGSalas I didn't add tests for this view since the deletion of the user is made completely in the user's backend by calling different methods of the platform.

You can maybe test your permissions at the very least.

File: eox_core/edxapp_wrapper/backends/users_j_v1.py (hunk 3 above)
Comment (magajh) [reply eox_core/edxapp_wrapper/backends/users_j_v1.py]: you're right, this was showing the response in a weird format.

**Comment by magajh** [comment]
> @magajh
> 
> > @MoisesGSalas I didn't add tests for this view since the deletion of the user is made completely in the user's backend by calling different methods of the platform.
> 
> You can maybe test your permissions at the very least.

@MoisesGSalas  Done! I added the tests for the permissions

File: eox_core/edxapp_wrapper/backends/users_j_v1.py [hunk 4]
```diff
@@ -180,6 +189,73 @@ def get_edxapp_user(**kwargs):
...
+                Registration.objects.filter(user=user).delete()
+
+                # Delete OAuth tokens associated with the user.
+                retire_dot_oauth2_models(user)
+
+                # Delete user signup source object
+                signup_sources[0].delete()
+
+                msg = "{user} has been removed".format(user=user_response)
+                status_code = status.HTTP_204_NO_CONTENT
```
[the remark this answers is not in the document]
Comment (magajh) [reply eox_core/edxapp_wrapper/backends/users_j_v1.py]: Done

File: eox_core/edxapp_wrapper/backends/users_j_v1.py [hunk 5]
```diff
@@ -180,6 +189,73 @@ def get_edxapp_user(**kwargs):
...
+                    signup_source.delete()
+
+                    msg = "{user} has more than one signup source. The signup source from the site {site} has been deleted".format(
+                        user=user_response,
+                        site=site,
+                    )
+                    status_code = status.HTTP_200_OK
+    else:
+        msg = "{user} does not have a signup source on the site {site}".format(user=user_response, site=site)
+        status_code = status.HTTP_400_BAD_REQUEST
```
[the remark this answers is not in the document]
Comment (magajh) [reply eox_core/edxapp_wrapper/backends/users_j_v1.py]: Done

**Review by MoisesGSalas — APPROVED** [review]

**Pull request merged**